scripts/integration-test.sh

```diff
@@ -1,6 +1,531 @@
+#------------------------------------------------------------------------------
+function retry_command() {
+  name=$1
+  command_str=$2
+  count=0
+  while true
+  do
+    $command_str >/dev/null
+    if [ $? != 0 ]
+    then
+      echo "INFO: waiting for $name..."
+      if [ $count == 30 ]
+      then
+        cat $name.log
+        fatal 1 "$name timeout"
+      fi
+      sleep 5
+      count=$((count+1))
+    else
+      cat_log_on_log_error $name
+      echo "INFO: $name app is without errors"
+      break
+    fi
+  done
+}
+
+#------------------------------------------------------------------------------
+# send in a basename of a log file and a search term.
+# grep for the search term in the log file.
+#------------------------------------------------------------------------------
+function retry() {
+  name=$1
+  search=$2
+  count=0
+  while true
+  do
+    grep -q "$search" $name.log
+    if [ $? != 0 ]
+    then
+      echo "INFO: waiting for $name..."
+      if [ $count == 30 ]
+      then
+        cat $name.log
+        fatal 1 "$name timeout"
+      fi
+      sleep 5
+      count=$((count+1))
+    else
+      cat_log_on_log_error $name
+      echo "INFO: $name app is without errors"
+      break
+    fi
+  done
+}
+
+#------------------------------------------------------------------------------
+# setup and run the collector, processor and middleware
+# The collector should be configured using the 2015 method of having the
+# ability to collect multiple crash types using different end points.
+#    breakpad crashes on /submit
+#    pellet stove data on /pellet/submit
+#----------------------------------------------------------------------------
+function start_2015_socorro_apps() {
+  echo "INFO: starting up *** start_2015_socorro_apps *** "
+  echo "      * in this test we're using a collector2015 capable of accepting 2 types of crashes"
+  echo "      * the collector writes to both the filesystem using FSPermanentStorage & to RabbitMQ using the $rmq_normal_queue_name queue"
+  echo "      * the processor used Processor2015 with the MozillaProcessorAlgorithm2015"
+  socorro collector2015 \
+      --services.services_controller='[{"name": "collector", "uri": "/submit", "service_implementation_class": "socorro.collector.wsgi_breakpad_collector.BreakpadCollector2015"}, {"name": "pellet_stove", "uri": "/pellet/submit", "service_implementation_class": "socorro.collector.wsgi_generic_collector.GenericCollector"}]' \
+      --services.collector.storage.crashstorage_class=socorro.external.crashstorage_base.PolyCrashStorage \
+      --services.collector.storage.storage_classes='socorro.external.fs.crashstorage.FSPermanentStorage, socorro.external.rabbitmq.crashstorage.RabbitMQCrashStorage' \
+      --services.collector.storage.storage0.crashstorage_class=socorro.external.fs.crashstorage.FSPermanentStorage \
+      --services.collector.storage.storage1.crashstorage_class=socorro.external.rabbitmq.crashstorage.RabbitMQCrashStorage \
```

Comment:
Why do these two storage classes need to be defined again? Isn't this what line 222 already did?